### ISSUE TITLE
View / Panes / Console on Left or Right not working properly

### ISSUE THREAD

**Issue by gtritchie** [rstudio/rstudio#7246]
### System details

    RStudio Edition : Desktop or Server
    RStudio Version : 1.3.1054 (also 1.4 and as far back as 1.3.624; worked in 1.3.623)
    OS Version      : any
    R Version       : any

### Steps to reproduce the problem
- startup IDE in a clean default state, with no files open in source pane (for best chance of repro, delete all previous settings including Qt-related settings if using Desktop)
- in View / Panes menu, note checkmark on "Console on Left" menu item
- select View / Panes / Console on Right

### Describe the problem in detail
- an editor document is opened (which is consistent with earlier versions) but the Console remains in lower-left corner
- View / Panes / Console on Right is now checked in the menu despite Console not being on right
- Problem persists even after quit and restart (i.e. Console in lower-left, menu says it's on right)
- errors logged:

```
Invalidating tabset config (Presentation index)
Pane config is not valid
```

### Describe the behavior you expected
- Untitled editor document created
- Console panel switches places with Environment panel
- Menu checkbox matches actual visual state including after a restart

### Investigation notes
- seems to have been introduced in build 1.3.624, either by #5788 or #5789
- attempts to undo those changes in current 1.3 codebase met with inconsistent results (seems like unrelated GWT caching issues which I've hit in other attempts to manually bisect and rebuild GWT code lately; strangely end up having to reboot my Mac to get back to a predictable state of affairs!)
- still actively investigating

- [x] I have read the guide to submitting good bug reports at https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports . 
- [x] I have installed the latest version of RStudio and confirmed that the issue still persists.
- [x] If I am reporting a RStudio crash, I have included a diagnostics report. https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by gtritchie** [comment]
This is proving unpleasant! For me it repros consistently with official builds, but local dev builds vary in whether they exhibit the issue.

**Comment by R-Hannibal** [comment]
Verified Fixed on: RStudio Version - 1.3.1073-1, R Version - 4.0.2 - Red Hat 8.2 RSP environment (accessed via Windows Desktop/Chrome). 

Comparison tested between current preview (1.3.1073-1) and current stable (1.3.1056), Console on Right / Console on Left are now toggling correctly in the new build (console can be successfully switched back and forth).

Also did some exploratory testing around this area as outlined by @gtritchie's QA Notes (tried combining changes in tab layout with the presence or absence of the Presentation tab, Closing/Opening sessions, as well as switching Console Left/Right). The change seems stable and the functionality tested around this change all seemed appropriate.
